### PR TITLE
Move VAOS cert file read to configuration and memoize it

### PR DESCRIPTION
**Pull request by kreek**
## Description of change
Moves the VAOS cert that is used for signing JWT requests to the base service configuration file.

## Testing done
unit tests

## Testing planned
staging integration tests

## Acceptance Criteria (Definition of Done)
does not break things and loads the cert file only once per server instance

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/vaos/app/services/vaos/configuration.rb [hunk 1]
```diff
@@ -14,6 +14,10 @@ def service_name
       'VAOS'
     end
 
+    def cert_key
+      @key ||= OpenSSL::PKey::RSA.new(File.read(Settings.va_mobile.key_path))
+    end
```
Comment (omgitsbillryan) [inline modules/vaos/app/services/vaos/configuration.rb]: `VAOS::Configuration` is a singleton, and therefore any instance variables in the singleton are cached for the life of the app... am I following that right? Just want to make sure this doesn't get loaded over and over again.

Secondly, not very important, but I think `key` is a more succinct name for this method.

Comment (kreek) [reply modules/vaos/app/services/vaos/configuration.rb]: yes it's a singleton, fairly long inheritance chain but the base class is `Common::Client::Configuration::Base`

Comment (saneshark) [reply modules/vaos/app/services/vaos/configuration.rb]: I can confirm, it is a singleton and therefore should not be loaded over and over again.

File: modules/vaos/app/services/vaos/configuration.rb [hunk 2]
```diff
@@ -14,6 +14,10 @@ def service_name
       'VAOS'
     end
 
+    def key
```
Comment (saneshark) [inline modules/vaos/app/services/vaos/configuration.rb]: NITPICK: possibly consider calling this "cert" or "private_key" as "key" is kind of generic and could be confused with "hash key"

File: modules/vaos/app/services/vaos/jwt.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def initialize(user)
     end
 
     def token
-      ::JWT.encode(payload, rsa_private, 'RS512')
+      ::JWT.encode(payload, Configuration.instance.key, 'RS512')
```
Comment (saneshark) [inline modules/vaos/app/services/vaos/jwt.rb]: NITPICK: is Configuration as clear as "VAOS::Configuration?" I guess so.

**Comment by saneshark** [comment]
Also, just FYI, I have a PR in the works where I'm trying to remove "user_configuration" so it's nice that you did this as part of the main "configuration" and not "user_configuration"

File: modules/vaos/app/services/vaos/configuration.rb (hunk 2 above)
Comment (kreek) [reply modules/vaos/app/services/vaos/configuration.rb]: Used to be called `cert_key` but changed it as I liked @omgitsbillryan suggestion to make it more succinct 😃Variable naming ~fight~ discussion... go! 🥊

Comment (kreek) [reply modules/vaos/app/services/vaos/configuration.rb]: `rsa_key` makes sense to me

**Pull request merged**